ksqldb-engine/src/test/java/io/confluent/ksql/planner/plan/QueryFilterNodeTest.java

```diff
@@ -1676,4 +1688,277 @@ public void shouldThrowNotKeyColumnForBetween() {
+            metaStore,
+            ksqlConfig,
+            false,
+            plannerOptions);
+
+    // When:
+    final List<LookupConstraint> keys = filterNode.getLookupConstraints();
+
+    // Then:
+    assertThat(keys.size(), is(1));
+    assertThat(keys.get(0), instanceOf(NonKeyConstraint.class));
+  }
+
+  @Test
+  public void shouldReturnKeyConstraintStringGreater() {
+    // Given:
+    final LogicalSchema schema = LogicalSchema.builder()
+            .keyColumn(ColumnName.of("K1"), SqlTypes.STRING)
+            .valueColumn(ColumnName.of("C1"), SqlTypes.INTEGER)
+            .build();
+    when(source.getSchema()).thenReturn(schema);
+    final Expression expression = new ComparisonExpression(
+            Type.GREATER_THAN,
+            new UnqualifiedColumnReferenceExp(ColumnName.of("K1")),
+            new StringLiteral("v1")
+    );
+    QueryFilterNode filterNode = new QueryFilterNode(
+            NODE_ID,
+            source,
+            expression,
+            metaStore,
+            ksqlConfig,
+            false,
+            plannerOptions
+    );
+
+    // When:
+    final List<LookupConstraint> keys = filterNode.getLookupConstraints();
+
+    // Then:
+    assertThat(keys.size(), is(1));
+    assertThat(keys.get(0), instanceOf(KeyConstraint.class));
+    final KeyConstraint keyConstraint = (KeyConstraint) keys.get(0);
+    assertThat(keyConstraint.getKey(), is(GenericKey.genericKey("v1")));
+  }
+
+  @Test
+  public void shouldReturnKeyConstraintString() {
+    // Given:
+    when(plannerOptions.getTableScansEnabled()).thenReturn(true);
+    final LogicalSchema schema = LogicalSchema.builder()
+            .keyColumn(ColumnName.of("K1"), SqlTypes.STRING)
+            .valueColumn(ColumnName.of("C1"), SqlTypes.INTEGER)
+            .build();
+    when(source.getSchema()).thenReturn(schema);
+    final Expression keyExp1 = new ComparisonExpression(
+            Type.GREATER_THAN,
+            new UnqualifiedColumnReferenceExp(ColumnName.of("K1")),
+            new StringLiteral("v1")
+    );
+    final Expression keyExp2 = new ComparisonExpression(
+            Type.EQUAL,
+            new UnqualifiedColumnReferenceExp(ColumnName.of("K1")),
+            new StringLiteral("v2")
+    );
+    final Expression expression = new LogicalBinaryExpression(
+            LogicalBinaryExpression.Type.AND,
+            keyExp1,
+            keyExp2
+    );
+    QueryFilterNode filterNode = new QueryFilterNode(
+            NODE_ID,
+            source,
+            expression,
+            metaStore,
+            ksqlConfig,
+            false,
+            plannerOptions);
+
+    // When:
+    final List<LookupConstraint> keys = filterNode.getLookupConstraints();
+
+    // Then:
+    assertThat(keys.size(), is(1));
+    assertThat(keys.get(0), instanceOf(KeyConstraint.class));
+    final KeyConstraint keyConstraint = (KeyConstraint) keys.get(0);
+    assertThat(keyConstraint.getKey(), is(GenericKey.genericKey("v2")));
```

Comment:
Would be great to check the operator here and elsewhere too just to be sure we didn't accidentally swap them with other key values.

Reply:
Done